util/serial/src/lib.rs

```diff
@@ -123,6 +133,64 @@ mod json_u64 {
 #[cfg(feature = "serde_with")]
 pub use json_u64::JsonU64;
 
+#[cfg(feature = "serde_with")]
+mod json_u128 {
+
+    use super::*;
+
+    /// Represents u128 using string, when serializing to Json
+    /// Javascript integers are not 64 bit, and so it is not really proper json.
+    /// Using string avoids issues with some json parsers not handling large
+    /// numbers well.
+    ///
+    /// This does not rely on the serde-json arbitrary precision feature, which
+    /// (we fear) might break other things (e.g. https://github.com/serde-rs/json/issues/505)
+    #[derive(Clone, Copy, Debug, Default, Deserialize, Eq, PartialEq, Hash, Serialize)]
+    #[serde(transparent)]
+    pub struct JsonU128(#[serde(with = "serde_with::rust::display_fromstr")] pub u128);
+
+    impl From<&u128> for JsonU128 {
+        fn from(src: &u128) -> Self {
+            Self(*src)
+        }
+    }
+
+    impl From<&u64> for JsonU128 {
+        fn from(src: &u64) -> Self {
+            Self(*src as u128)
+        }
+    }
+
+    impl From<&JsonU128> for u128 {
+        fn from(src: &JsonU128) -> u128 {
+            src.0
+        }
+    }
+
+    impl From<JsonU128> for u128 {
+        fn from(src: JsonU128) -> u128 {
+            src.0
+        }
+    }
+
+    impl From<JsonU128> for f64 {
```

Comment:
What was the reason you wanted to have this conversion? Do we use f64 anywhere in the auditor?

Reply:
Fwiw, its generally my preference that lossy conversion is explicit rather than implicit, so id rather not fold this into a From conversion. Id rather make the programmer explicitly convert f64 to u128 and then to JsonU128